subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectCollectionExtensions.kt

```diff
@@ -28,6 +28,34 @@ import kotlin.reflect.KProperty
 import kotlin.reflect.full.safeCast
 
 
+/**
+ * Returns a collection containing the objects in this collection of the given type. Equivalent to calling
+ * {@code withType(type).all(configureAction)}
+ *
+ * @param S The type of objects to find.
+ * @param configuration The action to execute for each object in the resulting collection.
+ * @return The matching objects. Returns an empty collection if there are no such objects
+ * in this collection.
+ * @see [NamedDomainObjectCollection.withType]
+ */
+inline fun <reified S : Any> NamedDomainObjectCollection<in S>.withType(noinline configuration: S.() -> Unit): NamedDomainObjectCollection<S> =
+    withType().also { it.all(configuration) }
```

Comment:
:x: `configureEach`?

Reply:
@bamboo Not in this case. `withType` that takes an `Action` always uses the immediate configuration API (calling `all`).

This is the docs from `DomainObjectCollection`:
```java
    /**
     * Returns a collection containing the objects in this collection of the given type. Equivalent to calling
     * {@code withType(type).all(configureAction)}
     *
     * @param type The type of objects to find.
     * @param configureAction The action to execute for each object in the resulting collection.
     * @return The matching objects. Returns an empty collection if there are no such objects in this collection.
     */
    <S extends T> DomainObjectCollection<S> withType(Class<S> type, Action<? super S> configureAction);
```

Reply:
:x: Right. It would be inconsistent to deviate from that.

I would remove the extension until a similar method exists in the core Gradle API.

Reply:
@bamboo Can you elaborate on why you think this? Maybe I'm simply misunderstanding you.

Reply:
Makes sense to remove it for now.
`NamedDomainObjectCollection` only overrides `withType(Class)` from `DomainObjectCollection`

Reply:
@JLLeitschuh For two reasons:
1. the DSL sugar should encourage configuration avoidance whenever possible
2. `NamedDomainObjectCollection` only overrides `withType(Class)` not `withType(Class, Action)`